### PR TITLE
MOD renamed `axis` meta-Data field to `orientation

### PR DESCRIPTION
**Pull request by AndreasThenn**
related to #15

### CODE REVIEW

File: Source/Components/StackCluster.swift [hunk 1]
```diff
@@ -35,11 +35,15 @@ extension ClusterLayout {
...
             let preserveParentWidth = meta["preserve_parent_width"] as? Bool
-            let axisRawValue = meta["axis"] as? Int
-            let axis = axisRawValue.flatMap { UILayoutConstraintAxis(rawValue: $0) }
+            let orientationRawValue = meta["orientation"] as? String
+            var axis = UILayoutConstraintAxis.vertical
             let hexString = meta["background_color"] as? String
             let backgroundColor = hexString.flatMap { UIColor(hexString: $0) }
-
+            
+            if "horizontal" == orientationRawValue {
```
Comment (ktraunmueller) [inline Source/Components/StackCluster.swift]: We should maybe introduce an enum with (`String`-valued) raw values for this and use the failable initializer to map strings onto enumeration cases.

Something like this:
```
var axis = UILayoutConstraintAxis.vertical
if let orientationRawValue = meta["orientation"] as? String,
   let orientation = Orientation(rawValue: orientationRawValue) {
     axis = orientation.layoutConstraintAxis
}
```
with
```
public enum ClusterLayout {
    
    public enum Orientation: String {
        case horizontal
        case vertical
    }
}

extension ClusterLayout.Orientation {
    
    var layoutConstraintAxis: UILayoutConstraintAxis {
        switch self {
        case .horizontal: return .horizontal
        case .vertical: return .vertical
        }
    }
}
```

Comment (AndreasThenn) [reply Source/Components/StackCluster.swift]: I like your approach, I tried to get the String name of out of UILayoutConstraintAxis, but that did not work. I updated the PR

**Review by ktraunmueller — APPROVED** [review]

File: Source/Components/StackCluster.swift [hunk 2]
```diff
@@ -35,11 +35,14 @@ extension ClusterLayout {
             let title = meta["title"] as? String
             let spacing = (meta["spacing"] as? NSNumber)?.floatValue
             let preserveParentWidth = meta["preserve_parent_width"] as? Bool
-            let axisRawValue = meta["axis"] as? Int
-            let axis = axisRawValue.flatMap { UILayoutConstraintAxis(rawValue: $0) }
+            var axis = UILayoutConstraintAxis.vertical
+            if let orientationRawValue = meta["orientation"] as? String,
```
Comment (ChristianOrgler) [inline Source/Components/StackCluster.swift:39]: don't you need to adapt the example json as well?

**Review by ChristianOrgler — APPROVED** [review]
Please adapt any example .json files from axis to orientation.

File: Source/Components/StackCluster.swift (hunk 2 above)
Comment (AndreasThenn) [reply Source/Components/StackCluster.swift:39]: It already used `orientation`

**Review by mathiasAichinger — APPROVED** [review]

File: Source/Components/ClusterLayout.swift [hunk 3]
```diff
@@ -14,5 +14,17 @@ import Foundation
 ///
 /// See the documentation of the clusters for more informations and configuration options.
 public enum ClusterLayout {
-    /* empty ATM, extended in other files... at some point there will be something here */
+    public enum Orientation: String {
+        case horizontal
+        case vertical
+    }
```
Comment (joanromano) [inline Source/Components/ClusterLayout.swift]: I am not sure if this information belongs here: afaik the `ClusterLayout` public enum is an enumeration to add new clusters, and the orientation is something that is not used in all of them.

Plus, I don't find this enum useful to be exposed publicly. When would this be used? As it is right now is only used by the library to parse the `ComponentMeta` into a `ExpressibleByComponentMeta` stack config. I would keep this thing internal to the library somewhere else if you really need to use it (which maybe we don't really need it for now).

File: Source/Components/StackCluster.swift [hunk 4]
```diff
@@ -35,11 +35,14 @@ extension ClusterLayout {
...
+                let orientation = Orientation(rawValue: orientationRawValue) {
+                axis = orientation.layoutConstraintAxis
+            }
```
Comment (joanromano) [inline Source/Components/StackCluster.swift:42]: If it's only for this, I don't think a `Orientation` enum is helpful for now. I would remove it and do the parsing normally as with the other fields

**Review by joanromano — CHANGES_REQUESTED** [review]
Please update the `StackClusterTests` where using a dictionary (`ComponentMeta`); those tests are still using `axis` but they pass because .vertical is now the default value when parsing. Adding a new test with this new `orientation` String value would have helped here. And I would really rethink about the need of that new `Orientation` enum.

But more importantly, please update the JSON schema guide https://github.com/runtastic/Matrioska/blob/master/Documentation/JSON%20schema%20guide.md otherwise it's reflecting wrong information.

File: Source/Components/StackCluster.swift (hunk 4 above)
Comment (MP0w) [reply Source/Components/StackCluster.swift:42]: +1 is not something helpful for other Components

File: Source/Components/ClusterLayout.swift (hunk 3 above)
Comment (ChristianOrgler) [reply Source/Components/ClusterLayout.swift]: I agree 👍

Comment (ktraunmueller) [reply Source/Components/ClusterLayout.swift]: Right, good point.

File: Source/Components/StackCluster.swift (hunk 4 above)
Comment (ktraunmueller) [reply Source/Components/StackCluster.swift:42]: Agreed, no need to expose that. Maybe make it a private type (of `StackCluster`)?

Comment (joanromano) [reply Source/Components/StackCluster.swift:42]: If you really think it's useful to keep it for the parsing ok, yes I would keep it private.

Comment (AndreasThenn) [reply Source/Components/StackCluster.swift:42]: thx for the feedback, @joanromano I updated my PR

Comment (joanromano) [reply Source/Components/StackCluster.swift:42]: Thanks, please don't forget to update the tests and the JSON schema guide as mentioned above.

**Review by joanromano — CHANGES_REQUESTED** [review]
Please update `StackClusterTests`, I believe there are tests using `axis` in their `ComponentMeta` and they pass just because we test against the default value.

**Comment by AndreasThenn** [comment]
@joanromano thx, indeed one test just passed because it used default value. I modified it to use orientation and added a new one to not use the default value. I added a new one, because the horizontal orientation of the first one just looked really ugly. Please check if you're fine with the new setup.

**Review by joanromano — APPROVED** [review]

**Pull request merged**